contracts/Ibc.sol

```diff
@@ -10,9 +12,25 @@ struct IbcPacket {
     /// The sequence number of the packet on the given channel
     uint64 sequence;
     bytes data;
-    /// when packet times out, measured on remote chain
-    /// Only inbound packets support both timeout height and timestamp.
-    IbcTimeout timeout;
+    /// block height after which the packet times out
+    Height timeoutHeight;
+    /// block timestamp (in nanoseconds) after which the packet times out
+    uint64 timeoutTimestamp;
+}
+
+/// Height is a monotonically increasing data type
+/// that can be compared against another Height for the purposes of updating and
+/// freezing clients
+///
+/// Normally the RevisionHeight is incremented at each height while keeping
+/// RevisionNumber the same. However some consensus algorithms may choose to
+/// reset the height in certain conditions e.g. hard forks, state-machine
+/// breaking changes In these cases, the RevisionNumber is incremented so that
+/// height continues to be monitonically increasing even as the RevisionHeight
+/// gets reset
+struct Height {
```

Comment:
Just a note here, not a comment:

`Height` here is a general way to represent block chain height.
- When `Height` represents a cosmos height (e.g. LC's field for Polymer height), `revision_number` has a meaning.
- When `Height` represents a EVM height, `revision_number` is undefined.